presto-main/src/main/java/com/facebook/presto/operator/aggregation/Histogram.java

```diff
@@ -0,0 +1,167 @@
+
+    public static final int EXPECTED_SIZE_FOR_HASHING = 10;
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Count the number of times each value occurs";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type keyType = types.get("K");
+        Type valueType = BigintType.BIGINT;
+        Signature signature = new Signature(NAME, new MapType(keyType, valueType).getTypeSignature(), keyType.getTypeSignature());
+        InternalAggregationFunction aggregation = generateAggregation(keyType, valueType);
+        return new FunctionInfo(signature, getDescription(), aggregation);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type keyType, Type valueType)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(Histogram.class.getClassLoader());
+        List<Type> inputTypes = ImmutableList.of(keyType);
+        Type outputType = new MapType(keyType, valueType);
+        HistogramStateSerializer stateSerializer = new HistogramStateSerializer(keyType);
+        Type intermediateType = stateSerializer.getSerializedType();
+        MethodHandle inputFunction = INPUT_FUNCTION.bindTo(keyType);
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, outputType, inputTypes),
+                createInputParameterMetadata(keyType),
+                inputFunction,
+                null,
+                null,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                HistogramState.class,
+                stateSerializer,
+                new HistogramStateFactory(),
+                outputType,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, outputType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type keyType)
+    {
+        return ImmutableList.of(new ParameterMetadata(STATE),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, keyType),
+                new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(Type type, HistogramState state, Block key, int position)
+    {
+        TypedHistogram typedHistogram = state.get();
+        if (typedHistogram == null) {
+            typedHistogram = new TypedHistogram(type, EXPECTED_SIZE_FOR_HASHING);
+            state.set(typedHistogram);
+        }
+
+        long startSize = typedHistogram.getEstimatedSize();
+        try {
+            typedHistogram.add(position, key, 1L);
+        }
+        catch (ExceededMemoryLimitException e) {
```

Comment:
How can you get this exception? I don't see it thrown in the `TypedHistogram` code